app/models/client_opportunity_match.rb

```diff
@@ -457,9 +457,14 @@ def rejected!
     end
   end
 
-  def canceled!
+  def canceled! (contact_id:nil)
     self.class.transaction do
       update! active: false, closed: true, closed_reason: 'canceled'
+      MatchEvents::Cancelled.create!(
```

Comment:
`ClientOpportunityMatch` can have a `has_many :canceled_events` similar to `note_events`.  Then this can be written: 
```
canceled_events.create!(client_id: self.client_id, contact_id: contact_id)
```
This will allow it to "know" the match_id.